infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespaceFactory.java

```diff
@@ -42,15 +55,19 @@ public KubernetesNamespaceFactory(
       @Nullable @Named("che.infra.kubernetes.cluster_role_name") String clusterRoleName,
       KubernetesClientFactory clientFactory) {
     this.namespaceName = namespaceName;
-    this.isPredefined = !isNullOrEmpty(namespaceName);
+    this.isPredefined = !isNullOrEmpty(namespaceName) && hasNoPlaceholders(this.namespaceName);
     this.serviceAccountName = serviceAccountName;
     this.clusterRoleName = clusterRoleName;
     this.clientFactory = clientFactory;
   }
 
+  private boolean hasNoPlaceholders(String namespaceName) {
+    return NAMESPACE_NAME_PLACEHOLDERS.keySet().stream().noneMatch(namespaceName::contains);
```

Comment:
Is this really worth it for a map of size 2 that is not that likely to grow?

Reply:
alternatives?

Reply:
BTW. 3. Next one workspace id https://github.com/eclipse/che/issues/14585

Reply:
If you know cleaner solution, please tell. 

With `workspace.id` it will be a bit trickier because I've used `Subject` here as a value in map and it has no workspace id. But I want to solve that in the next task, not to bother with it now.

Reply:
@metlos is @sparkoo 's answer ok for you? Or you have some solution in mind?

Reply:
Yes, it seems that it's going to grow. resolving this.